src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3848,6 +3848,20 @@ public bool TryGetSimilarName(DName name, FieldNameKind fieldNameKind, out strin
                    comparer.Distance(similar) < (name.Value.Length / 3) + 3;
         }
 
+        public bool AggregateHasExpandedType()
+        {
+            var ret = false;
+
+            if (IsAggregate)
+            {
+                var record = ToRecord();
+
+                ret = record.GetAllNames(DPath.Root).Any(name => name.Type.IsExpandEntity);
+            }
```

Comment:
Will it work if the type is a record with a field being a record itself and having an expandEntity itself (ie. Is this recursive?)

Reply:
> Will it work if the type is a record with a field being a record itself and having an expandEntity itself
Something like `{name: "John", account: First(Accounts)}` will work.

> Is this recursive?
I don't believe it is. We can address this in a new issue since this would be beyond the scope of this one.